mondrian/src/main/java/mondrian/rolap/MemberCacheHelper.java

```diff
@@ -164,15 +164,8 @@ private List<RolapMember> findNamedChildrenInCache(
         {
             return null;
         }
-        filter(
-            children, new Predicate()
-            {
-                public boolean evaluate(Object member) {
-                    return childNames.contains(
-                        ((RolapMember) member).getName());
-                }
-            });
-        boolean foundAll = children.size() == childNames.size();
+
+        boolean foundAll = children.parallelStream().allMatch( member -> childNames.contains( member.getName() ));
         return !foundAll ? null : children;
```

Comment:
(not a change done under this development, but couldn't help to notice this)

Why are we negating just for the "if"?
I think it would be best to have the following instead:
`return foundAll ? children : null;`